bundler/lib/bundler/dependency.rb

```diff
@@ -159,5 +159,13 @@ def specific?
     rescue NoMethodError
       requirement != ">= 0"
     end
+
+    private
+
+    def default_force_ruby_platform
+      return false unless Bundler.current_ruby.truffleruby?
+
+      !%w[libv8 sorbet-static].include?(name)
```

Comment:
(reposting it here so we can discuss this on the PR more easily, vs a comment on a commit)
Could we use `match_gem?` here so we avoid duplicating the logic of https://github.com/oracle/truffleruby/blob/b131d3c146afe6bed0f119d0891b34ba68ac5deb/lib/mri/rubygems/platform.rb#L38-L47 ?
That list is subject to change and needs to remain in the truffleruby repo.

Reply:
This was just a WIP but I was planning to use the constant.

Reply:
Truffleruby wants to force the ruby platform for some specific gems, using this feature seems like a perfect use case for me.

Reply:
> Truffleruby wants to force the ruby platform for some specific gems, using this feature seems like a perfect use case for me.

Ah right, this would be a cleaner way to integrate than to override `match_gem?` in TruffleRuby, is that what you mean?
That sounds good in general.
I'd be more comfortable if this `default_force_ruby_platform` was a method on e.g. `Gem` (and it would be passed the gem name and the version, as the answer might depend on the version) and TruffleRuby could override that method in `rubygems/defaults/truffleruby.rb`

Reply:
> Is there are reason we can't use match_gem? here?

We can, but using this feature feels way better to me. As you can see in d3317b8192cc912f9219c84f2b6302548be1fe34, it allows us to remove all the hacks we need to keep just for truffleruby, and that we introduced because of causing regressions. I think this is much cleaner and less likely to cause issues.

Reply:
> I'd be more comfortable if this default_force_ruby_platform was a method on e.g. Gem (and it would be passed the gem name and ideally the version too, as the answer might depend on the version) and TruffleRuby could override that method in rubygems/defaults/truffleruby.rb

I think passing the version is overkill to me. We haven't run resolution at this point yet, so we only have Gemfile information here. We could use the version if it's explicit in the Gemfile, but I think it's an unnecessary complication. Let's stick to this. Regarding moving the method, I don't think this is a good fit as a global `Gem` method, and I don't want to pollute `Gem` more unnecessarily. Why will you want to override this method?

Reply:
I added a commit to use `Gem::Platform::REUSE_AS_BINARY_ON_TRUFFLERUBY`. I'm happy to backfill it for older versions of truffleruby that didn't have it.

Reply:
Yes, agreed having the version is not worth the trouble at this point, just the gem name is enough currently.

> Regarding moving the method, I don't think this is a good fit as a global Gem method, and I don't want to pollute Gem more unnecessarily. 

It could be somewhere else, e.g., on Gem::Platform or any other place.
An instance method on Bundler::Dependency feels weird to me if we ever need that logic anywhere else, but maybe we don't, WDYT?
I guess `Bundler::Dependency#force_ruby_platform` is already a good way to get the result of this check.
An advantage of having it on Bundler::Dependency is we have more information should we ever need it. Would be the same if it's a method somewhere else and it's passed the Bundler::Dependency instance, but then indeed there is no point to move the method.

> Why will you want to override this method?

Because I want some flexibility, for instance I'd think the list might depend on some TruffleRuby option, or depend on the platform, or might want to match gem names with a regexp (e.g., to handle potential forks of those gems, or some related gems which should all be installed as binary).
In other words, a constant like that does not feel like a proper API, a method seems much more flexible.
But I'm OK to override `Bundler::Dependency#default_force_ruby_platform` in TruffleRuby, so that could be the "extension point".

It also feels cleaner to me if TruffleRuby overrides the method and the default is just `false`, because then there are no truffleruby specifics here, except for a comment on this new method to mention it is overwritten by truffleruby (but possibly also by other Ruby implementations which want to override it in the future).

Reply:
> It also feels cleaner to me if TruffleRuby overrides the method and the default is just `false`, because then there are no truffleruby specifics here, except for a comment on this new method to mention it is overwritten by truffleruby (but possibly also by other Ruby implementations which want to override it in the future).

Ah, but that can't be because then existing TruffleRuby releases would just break with latest Bundler, I didn't think about that ealier.
OK, then using the constant on those older TruffleRuby versions seems unavoidable.
Are you OK if newer TruffleRuby releases override this method? Could you add a comment that the method might be overridden by Ruby implementations and e.g. truffleruby?

Reply:
I think you can send us a patch with whatever comment is needed when/if you need to override the method in the future